### ISSUE TITLE
Cannot read parameters for disparity: Feature not found: Scan3dPrincipalPointU

### ISSUE THREAD

**Issue by xh20** [roboception/rc_visard_opencv_example#3]
Hi, thanks for these example codes. They are very helpful. Just found a small issue. When I wanna check the disparity:
`./rc_visard_show_streams --disparity <deviceID>`
and got an error: `Cannot read parameters for disparity: Feature not found: Scan3dPrincipalPointU`.

I have also tried to use the chunk mode `rcg::setBoolean(node_map, "ChunkModeActive", true)`, and added "Chunk" into the name. It still reports `Feature not found: ChunkScan3dPrincipalPointU`. 

Eventually, I checked the code in rc_genicam_api, it uses `rcg::getFloat(nodeMap, "Scan3dPrincipalPointU")` directly, without specifying the values of `vmin`, `vmax` and `expection`. So I changed the lines in the image_receiver.cc file:
`c_y = rcg::getFloat(node_map, GC_SCAN_3D_PRINCIPAL_POINT_V, 0, 0, true);`
`c_x = rcg::getFloat(node_map, GC_SCAN_3D_PRINCIPAL_POINT_U, 0, 0, true));`
`coord_scale = rcg::getFloat(node_map, GC_SCAN_3D_COORDINATE_SCALE, 0, 0, true);`
`rcg::getFloat(node_map, GC_SCAN_3D_INVALID_DATA_VALUE, 0, 0, true);`
to
`c_y = rcg::getFloat(node_map, GC_SCAN_3D_PRINCIPAL_POINT_V);`
`c_x = rcg::getFloat(node_map, GC_SCAN_3D_PRINCIPAL_POINT_U);`
`coord_scale = rcg::getFloat(node_map, GC_SCAN_3D_COORDINATE_SCALE);`
`rcg::getFloat(node_map, GC_SCAN_3D_INVALID_DATA_VALUE)`
and it works now for both chunk and no-chunk cases. For the focal length f and baseline t, the code also specifies these values and returns something. I am not sure if I am in the right way, please take a look. Thanks!

**Comment by heikohimu** [comment]
That is very strange. The Scan3dPrincipalPointU and V parameter are always available. With your "fix", you avoid throwing an exception if the parameter is not found or cannot be accessed. Instead 0 is returned, which will result in an incorrect reconstruction.

What is the firmware version of your rc_visard?
Can you check the content of the GENICAM_GENTL64_PATH environment variable (in a shell under Linux with 'echo $GENICAM_GENTL64_PATH' and under Windows I guess with 'echo %GENICAM_GENTL64_PATH%')?

**Comment by heikohimu** [comment]
The most likely reason for the error might be that you did not specify the DeviceID of an rc_visard, but of a "normal" GenICam camera?

**Comment by xh20** [comment]
Thanks for your help! After updating the firmware to the newest one (24.04), the original code works well now!

$GENICAM_GENTL64_PATH is empty, I found the genicam in the installed rc_genicam_api folder. I guess it is integrated into the API package. The rcdiscover-gui works well. I found the firmware version is 1.0.8, probably too old. After updating the firmware, the code works. The ID should be correct, it is got by rcdiscover-gui.
